OpenRA.Game/Network/Connection.cs

```diff
@@ -314,6 +314,11 @@ void NetworkConnectionReceive()
 					AddPacket(new ReceivedPacket { FromClient = client, Data = buf });
 				}
 			}
+			catch (InvalidOperationException ex)
+			{
+				errorMessage = "Connection failed";
+				Log.Write("client", ex.Message);
```

Comment:
But `ex.Message` is also logged in the case below.

Reply:
yes it is, but since you took the time to throw a well crafted and meaningful `InvalidOperationException` on 300, why would you log it generically?!?... a lot of crappy things can happen that would fall under the `Exception` umbrella (failures reading the socket, closed connections, `notimplementedexception` on line 313 and others) but since you decided to single that one out, I believe it deserved a singled-out treatment...
but if you don't like it, I can roll it back and let it be generically handled...